Build failure: python3Packages.stringzilla on aarch64-linux

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix build 'nixpkgs#python3Packages.stringzilla'

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310276484

### Relevant log output

```console
In file included from python/stringzilla.c:64:
include/stringzilla/stringzilla.h:272:8: error: unknown type name ‘sigjmp_buf’; did you mean ‘jmp_buf’?
  272 | static sigjmp_buf sz_mrs_test_jump_buffer_;
      |        ^~~~~~~~~~
      |        jmp_buf
include/stringzilla/stringzilla.h: In function ‘sz_mrs_test_sigill_handler_’:
include/stringzilla/stringzilla.h:275:16: error: passing argument 1 of ‘siglongjmp’ makes pointer from integer without a cast [-Wint-conversion]
  275 |     siglongjmp(sz_mrs_test_jump_buffer_, 1);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
      |                |
      |                int
In file included from /nix/store/kpwwc8ml9cvzv24glz7h2vc5xmdrab96-glibc-2.40-66-dev/include/features.h:511,
                 from /nix/store/kpwwc8ml9cvzv24glz7h2vc5xmdrab96-glibc-2.40-66-dev/include/fcntl.h:25,
                 from python/stringzilla.c:33:
/nix/store/kpwwc8ml9cvzv24glz7h2vc5xmdrab96-glibc-2.40-66-dev/include/bits/setjmp2.h:31:13: note: expected ‘struct __jmp_buf_tag *’ but argument is of type ‘int’
   31 | extern void __REDIRECT_NTHNL (siglongjmp,
      |             ^~~~~~~~~~~~~~~~
include/stringzilla/stringzilla.h: In function ‘sz_capabilities_implementation_arm_’:
include/stringzilla/stringzilla.h:313:22: error: storage size of ‘action_new’ isn’t known
  313 |     struct sigaction action_new, action_old;
      |                      ^~~~~~~~~~
include/stringzilla/stringzilla.h:313:34: error: storage size of ‘action_old’ isn’t known
  313 |     struct sigaction action_new, action_old;
      |                                  ^~~~~~~~~~
include/stringzilla/stringzilla.h:315:5: error: implicit declaration of function ‘sigemptyset’ [-Wimplicit-function-declaration]
  315 |     sigemptyset(&action_new.sa_mask);
      |     ^~~~~~~~~~~
include/stringzilla/stringzilla.h:319:9: error: implicit declaration of function ‘sigaction’ [-Wimplicit-function-declaration]
  319 |     if (sigaction(SIGILL, &action_new, &action_old) == 0) {
      |         ^~~~~~~~~
include/stringzilla/stringzilla.h:320:13: error: implicit declaration of function ‘sigsetjmp’; did you mean ‘_setjmp’? [-Wimplicit-function-declaration]
  320 |         if (sigsetjmp(sz_mrs_test_jump_buffer_, 1) == 0) {
      |             ^~~~~~~~~
      |             _setjmp
include/stringzilla/stringzilla.h:313:34: warning: unused variable ‘action_old’ [-Wunused-variable]
  313 |     struct sigaction action_new, action_old;
      |                                  ^~~~~~~~~~
include/stringzilla/stringzilla.h:313:22: warning: unused variable ‘action_new’ [-Wunused-variable]
  313 |     struct sigaction action_new, action_old;
      |                      ^~~~~~~~~~
error: command '/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/gcc' failed with exit code 1
```

### Additional context

https://github.com/ashvardanian/StringZilla/pull/262 looks relevant.

### System metadata

 - system: `"aarch64-linux"`
 - host os: `Linux 6.16.7, NixOS, 25.11 (Xantusia), 25.11.20250913.c23193b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - nixpkgs: `/nix/store/lvwgkdy9nrki8qcrdsqnpfrk7562m1dc-source`

### Notify maintainers

@dotlambda

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/stringzilla/default.nix
+++ b/pkgs/development/python-modules/stringzilla/default.nix
@@ -21,6 +21,10 @@ buildPythonPackage rec {
     hash = "sha256-0CIekVxChvH912vFnBF2FR1YyIpxi3SD7KhBlh7yFGA=";
   };
 
+  # Define _POSIX_C_SOURCE to enable POSIX signal handling for ARM capability detection
+  # See: https://github.com/ashvardanian/StringZilla/pull/263
+  env.NIX_CFLAGS_COMPILE = "-D_POSIX_C_SOURCE=200809L";
+
   postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
     # error: unsupported option '-mfloat-abi=' for target 'aarch64-apple-darwin'
     substituteInPlace setup.py \

--- a/pkgs/development/python-modules/stringzilla/default.nix
+++ b/pkgs/development/python-modules/stringzilla/default.nix
@@ -46,6 +46,9 @@ buildPythonPackage rec {
     description = "SIMD-accelerated string search, sort, hashes, fingerprints, & edit distances";
     homepage = "https://github.com/ashvardanian/stringzilla";
     license = lib.licenses.asl20;
-    maintainers = with lib.maintainers; [ dotlambda ];
+    maintainers = with lib.maintainers; [
+      aciceri
+      dotlambda
+    ];
   };
 }

